### ISSUE TITLE
Wrong operator in reposync check

### ISSUE THREAD

**Issue by scroogie** [cobbler/cobbler#1586]
Since b98af58a75c7c833073ade688829362035203710, cobbler check complains about reposync missing if either dnf or reposync is missing. Instead of 

if not os.path.exists("/usr/bin/dnf") or not os.path.exists("/usr/bin/reposync"): 

it should actually be 

if not os.path.exists("/usr/bin/dnf") and not os.path.exists("/usr/bin/reposync"): 

because sync can now use both dnf or reposync. 
Probably, the error message should also be changed to include dnf.

**Comment by scroogie** [comment]
And by the way, checking /usr/bin/dnf doesn't seem to be enough, as reposync is provided by dnf-plugins-core.

**Comment by AlexVRud** [comment]
```console
cobbler$ grep -rn -A 5 -B 5 "bin/dnf" .
./cobbler/action_reposync.py-272-        """
./cobbler/action_reposync.py-273-        Determine reposync command
./cobbler/action_reposync.py-274-        """
./cobbler/action_reposync.py-275-
./cobbler/action_reposync.py-276-        cmd = None                # reposync command
./cobbler/action_reposync.py:277:        if os.path.exists("/usr/bin/dnf"):
./cobbler/action_reposync.py:278:            cmd = "/usr/bin/dnf reposync"
./cobbler/action_reposync.py-279-        elif os.path.exists("/usr/bin/reposync"):
./cobbler/action_reposync.py-280-            cmd = "/usr/bin/reposync"
./cobbler/action_reposync.py-281-        else:
./cobbler/action_reposync.py-282-            # warn about not having yum-utils.  We don't want to require it in the package because
./cobbler/action_reposync.py-283-            # Fedora 22+ has moved to dnf.
--
./cobbler/action_reposync.py-411-
./cobbler/action_reposync.py-412-            # older yumdownloader sometimes explodes on --resolvedeps
./cobbler/action_reposync.py-413-            # if this happens to you, upgrade yum & yum-utils
./cobbler/action_reposync.py-414-            extra_flags = self.settings.yumdownloader_flags
./cobbler/action_reposync.py-415-            cmd = ""
./cobbler/action_reposync.py:416:            if os.path.exists("/usr/bin/dnf"):
./cobbler/action_reposync.py:417:                cmd = "/usr/bin/dnf download"
./cobbler/action_reposync.py-418-            else:
./cobbler/action_reposync.py-419-                cmd = "/usr/bin/yumdownloader"
./cobbler/action_reposync.py-420-            cmd = "%s %s %s --disablerepo=* --enablerepo=%s -c %s --destdir=%s %s" % (cmd, extra_flags, use_source, repo.name, temp_file, dest_path, " ".join(repo.rpm_list))
./cobbler/action_reposync.py-421-
./cobbler/action_reposync.py-422-        # now regardless of whether we're doing yumdownloader or reposync
--
./cobbler/action_check.py-142-        if self.checked_family == "debian":
./cobbler/action_check.py-143-            return
./cobbler/action_check.py-144-
./cobbler/action_check.py-145-        if not os.path.exists("/usr/bin/createrepo"):
./cobbler/action_check.py-146-            status.append(_("createrepo package is not installed, needed for cobbler import and cobbler reposync, install createrepo?"))
./cobbler/action_check.py:147:        if not os.path.exists("/usr/bin/dnf") or not os.path.exists("/usr/bin/reposync"):
./cobbler/action_check.py-148-            status.append(_("reposync is not installed, need for cobbler reposync, install/upgrade yum-utils?"))
./cobbler/action_check.py-149-        if not os.path.exists("/usr/bin/yumdownloader"):
./cobbler/action_check.py-150-            status.append(_("yumdownloader is not installed, needed for cobbler repo add with --rpm-list parameter, install/upgrade yum-utils?"))
./cobbler/action_check.py-151-        if self.settings.reposync_flags.find("-l"):
./cobbler/action_check.py-152-            if self.checked_family in ("redhat", "suse"):
```
May be need:
```python
        if not os.path.exists("/usr/bin/dnf") and not os.path.exists("/usr/bin/reposync"):
            status.append(_("dnf or reposync is not installed, need for cobbler reposync, install/upgrade yum-utils/dnf?"))
        if not os.path.exists("/usr/bin/dnf") and not os.path.exists("/usr/bin/yumdownloader")
            status.append(_("dnf or yumdownloader or dnf is not installed, needed for cobbler repo add with --rpm-list parameter, install/upgrade yum-utils/dnf?"))

```

**Comment by opoplawski** [comment]
yeah, not really sure the best way to check for dnf-plugins-core/dnf reposync.  Suggestions welcome.

**Comment by jmaas** [comment]
Thanks for reporting this fix will be in the next release 2.6.11
